File: app/src/main/java/com/example/bakerystore/models/ProductResponse.kt
package com.example.bakerystore.models

data class ProductResponse(
    val productId: Int,
    val productName: String,
    val description: String?,
    val price: Double,
    val imageUrl: String?,
    val stock: Int,
    val categoryId: Int?,
    val categoryName: String?,
    val createdAt: String?,
    val reviewCount: Int,
    val averageRating: Double
)

File: backend/BakeryStoreApi/DTOs/ProductResponse.cs
namespace BakeryStoreApi.DTOs
{
    public class ProductResponse
    {
        public int ProductId { get; set; }

        public string ProductName { get; set; } = string.Empty;

        public string? Description { get; set; }

        public decimal Price { get; set; }

        public string? ImageUrl { get; set; }

        public int Stock { get; set; }

        public int? CategoryId { get; set; }

        public string? CategoryName { get; set; }

        public DateTime CreatedAt { get; set; }

        public int ReviewCount { get; set; }

        public double AverageRating { get; set; }
    }
}